src/components/DynamicFormContainer/DataFileButton/DataFileButton.tsx

```diff
@@ -1,24 +1,31 @@
 import { Button } from "@govtechsg/tradetrust-ui-components";
 import React, { FunctionComponent, useState } from "react";
 import { useDropzone } from "react-dropzone";
-import { readFileAsJson } from "../../../common/utils";
+import { readFileAsJson, readFileAsCsv } from "../../../common/utils";
 import { getLogger } from "../../../utils/logger";
 import { ErrorAlert } from "../../Alert";
+import { useFormsContext } from "../../../common/context/forms";
 
 const { stack } = getLogger("DataFileButton");
 interface DataFileButton {
   onDataFile: (dataFile: any) => void; // eslint-disable-line @typescript-eslint/no-explicit-any
 }
 
 export const DataFileButton: FunctionComponent<DataFileButton> = ({ onDataFile }) => {
+  const { currentFormTemplate } = useFormsContext();
   const [error, setError] = useState(false);
   const onDrop = async (files: File[]): Promise<void> => {
     try {
       const file = files[0];
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      const config = await readFileAsJson<any>(file);
+      let data = null;
+      if (file.name.indexOf(".csv") > 0) {
```

Comment:
instead of overloading the same button i think it would be clearer to the user and also cleaner in the code if they are separate buttons clearly labelled and also with their own handlers, what do you think?

Reply:
In the current website, the data upload button can accept both JSON and CSV files. If we separate the button, they may require us to display separate buttons (upload JSON data file, upload CSV data file) in the document creator.